net/tstun/tap_linux.go

```diff
@@ -91,7 +95,7 @@ const (
 
 // handleTAPFrame handles receiving a raw TAP ethernet frame and reports whether
 // it's been handled (that is, whether it should NOT be passed to wireguard).
-func (t *Wrapper) handleTAPFrame(ethBuf []byte) bool {
+func (t *tapDevice) handleTAPFrame(ethBuf []byte) bool {
```

Comment:
moved these to be on the tapDevice instead of Wrapper as the circular dep made it harder to reason about.